policies/lib/tree/30_generic_methods/file_from_local_source_recursion.cf

```diff
@@ -44,31 +44,35 @@ bundle agent file_from_local_source_recursion(source, path, recursion)
       # depth_search prevents from copying file/symlink
       # so we need to detect if we want to copy a folder or a file
       # If we check if it is a file, the class will always be true
-      # event if was in fact a directory. Then, we test if it is a dir.
-      "is_dir_copy"  expression => isdir("${path}");
+      # even if was in fact a directory. Then, we test if it is a dir.
+      "is_dir_copy"  expression => isdir("${source}");
       # We won't be able to know if it is really a directory
       # or not before the first iteration
-      "iteration_2"  expression => "iteration_1";
-      "iteration_1"  expression => "any";
+      "pass3"  expression => "pass2";
+      "pass2"  expression => "pass1";
+      "pass1"  expression => "any";
 
   files:
-   !is_dir_copy::
+   pass3.!is_dir_copy::
       "${path}"
         copy_from    => ncf_local_cp_method("${source}", "digest"),
-        classes      => classes_generic_two("${report_data.method_id}", "${class_prefix}");
+        classes      => classes_generic("${report_data.method_id}");
 
-   is_dir_copy::
+   pass3.is_dir_copy::
       "${path}"
         copy_from    => ncf_local_cp_method("${source}", "digest"),
         depth_search => recurse("${recursion}"),
-        classes      => classes_generic_two("${report_data.method_id}", "${class_prefix}");
+        classes      => classes_generic("${report_data.method_id}");
```

Comment:
no generic_two because you sanitize once and you copy ? or else you would have to sanitize twice ?

Reply:
I would have to sanitize twice, which is a pain